src/sentry/tasks/auth.py

```diff
@@ -112,3 +137,46 @@ def _remove_2fa_non_compliant_member(member, org, actor=None, actor_key=None, ip
             context=email_context,
         )
         message.send_async([member.email])
+
+
+@instrumented_task(
+    name="sentry.tasks.remove_2fa_non_compliant_members",
+    queue="auth",
+    default_retry_delay=60 * 5,
+    max_retries=5,
+)
+def remove_2fa_non_compliant_members(org_id, actor_id=None, actor_key_id=None, ip_address=None):
+    TwoFactorComplianceTask().remove_non_compliant_members(
+        org_id, actor_id, actor_key_id, ip_address
+    )
+
+
+class VerifiedEmailComplianceTask(OrganizationComplianceTask):
+    log_label = "verified email"
+
+    def is_compliant(self, member: OrganizationMember) -> bool:
+        return UserEmail.get_primary_email(member.user).is_verified
+
+    def call_to_action(self, member: OrganizationMember):
+        user = member.user
+        if isinstance(user, User):
+            email = UserEmail.get_primary_email(user).email
+
+            # TODO: Need to send different email with additional info
+            #  that the user has been removed from the org and needs
+            #  to redo the invite flow?
+            user.send_confirm_email_singular(email)
```

Comment:
wonder if we can do the changes to allow for restricted members prior to this going live. i think users are going to have a hard time with this flow unless we combine the verify email /invite into the same email

Reply:
Prior to going live, sounds very preferable.

If our arm were twisted, we could at least include the verification link and invite link in the same email.

Now that I write that out, I realize that the *best* UX would be to click a single link that both verifies the email and rejoins you to the organization. It's equivalent from a security perspective, of course. Back-end support for that would be a bit of scope creep, though.